### PR TITLE
refactor: creates modifiers for intro block

### PR DESCRIPTION
**Pull request by MarPostovik**
## Description

In our project, we have three pages, all of which have an `intro` block. But they have some differences such as `padding` and `max-width`. I created modifiers for block `intro` and for its element `intro__title` to list those changes in separate files.

## Task Link

[Link to Figma](https://www.figma.com/file/TYlLseXoMXDflJYbU1DGHa/HTML-2-%2F-%D0%9F%D0%B8%D0%BD%D0%BA?type=design&node-id=1226-717&mode=design&t=KPFxn9IitFVV9Nte-0)

## Changes Made

- Created modifiers for `intro`  and `intro__title` (Listed below in usage)
- Placed in the files for those modifiers styles that are specific for every page
- Removed  styles that are only needed for page `index.html`, from the `intro.css`. (These styles now are in `intro_page_index.css`).
- Removed styles that are only needed for `form.html` from `intro__title.css`. (Now these styles are in `intro__title_page_form.css`)
- Added classes with new modifiers on pages `index.html` and `form.html`.
- Updated `index.css` and `form.css` with imports for new modifiers.

## Usage

Here is a quick instruction on how to use the modifiers.

Modifiers for block `intro`: 

- `intro_page_index`
- `intro_page_form`
- `intro_page_catalog`

They should be used along with the class `intro` where common styles for all intro blocks are placed.

Modifiers for element `intro__title`:

- `intro__title_page_form`
- `intro__title_page_catalog`

They should be used along with the class `intro__title`.

Example:

```html
<div class="intro intro_page_form">
        <h1 class="intro__title intro__title_page_form">Your title</h1>
 </div>
``` 
I've already added those modifiers on pages `index.html` and `form.html` but you may need this information for page `catalog.html`

## Checklist

- [X] I've validated my HTML with [Nu Html Checker](https://validator.w3.org/nu/), and it outputs no errors
- [X] I've checked my HTML with [HTML Tree Generator](https://yoksel.github.io/html-tree/en/), and it outputs no errors
- I've ensured that the layout matches the mockup for the following control points:
  - [X] 1200px
  - [X] 660px
  - [X] 320px

### CODE REVIEW

File: blocks/intro/__title/_page/intro__title_page_catalog.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  max-width: 357px;
+}
+@media (max-width: 1199px){
+  .intro__title_page_catalog {
+    padding: 177px 0 130px;
+  }
+}
+@media (max-width: 659px){
+  .intro__title_page_catalog {
+    max-width: 172px;
```
Comment (naumch1k) [inline blocks/intro/__title/_page/intro__title_page_catalog.css:11]: @MarPostovik While we've used `max-width` primarily for line breaking on larger screens to match the design, our priority on mobile devices is ensuring text _elements aren't too close to the device edges_. Setting the max width to `280px` for this element will be just enough to guarantee 20px space on both the right and left sides, accommodating even the smallest screens we need to consider, such as 320px wide. 

I recommend limiting the width of this title to `280px`, similar to the `.intro__title_page_form element`. Note that the visuals of this section at 320px won't change 😀

**Review by naumch1k — CHANGES_REQUESTED** [review]
@MarPostovik Thank you for taking the initiative to refactor the intro section and extending it to cover future pages where it will be used! It's genuinely commendable to witness your dedication to understanding the project deeply and your skill in planning ahead 🌻

I have left a few suggestions and comments for improvements below.

Feel free to reach out if you need any help ❤️

File: blocks/intro/intro.css [hunk 2]
```diff
@@ -1,26 +1,17 @@
 .intro{
```
[the remark this answers is not in the document]
Comment (MarPostovik) [reply blocks/intro/intro.css:1]: Yes, they don't affect anything. I removed them.

File: blocks/intro/intro.css [hunk 3]
```diff
@@ -1,26 +1,17 @@
...
-    .intro{
-        height: 729px;
-    }
-}
-
 @media(max-width:659px){
     .intro{
         background-image: url(../../images/intro-background-mobile.png);
-        height: 733px;
         background-size: cover;
```
[the remark this answers is not in the document]
Comment (MarPostovik) [reply blocks/intro/intro.css:11]: Done ✅

File: blocks/intro/__title/_page/intro__title_page_catalog.css [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    padding: 98px 0 78px 0;
```
[the remark this answers is not in the document]
Comment (MarPostovik) [reply blocks/intro/__title/_page/intro__title_page_catalog.css:12]: Done ✅ Thank you for the suggestion. ❤️

File: blocks/intro/_page/intro_page_index.css [hunk 5]
```diff
@@ -0,0 +1,9 @@
+.intro_page_index{
+  background-position: bottom center;
+}
+
+@media(max-width: 659px){
+  .intro_page_index{
+    padding: 0 0 270px 0;
```
[the remark this answers is not in the document]
Comment (MarPostovik) [reply blocks/intro/_page/intro_page_index.css]: Done ✅

File: blocks/intro/__title/_page/intro__title_page_catalog.css (hunk 1 above)
Comment (MarPostovik) [reply blocks/intro/__title/_page/intro__title_page_catalog.css]: done ✅

**Review by naumch1k — APPROVED** [review]
@MarPostovik, thank you for incorporating all the suggestions and improvements! Your code looks great to me ❤️‍🔥

Excellent work, as always!

**Pull request merged**